CONTRIBUTING.md

```diff
@@ -1,331 +1,444 @@
 # Contribution Guidelines
 
-## Table of Contents
-
-- [Contribution Guidelines](#contribution-guidelines)
-  - [Table of Contents](#table-of-contents)
-  - [Introduction](#introduction)
-  - [Bug reports](#bug-reports)
-  - [Discuss your design](#discuss-your-design)
-  - [Testing redux](#testing-redux)
-  - [Vendoring](#vendoring)
-  - [Translation](#translation)
-  - [Building Gitea](#building-gitea)
-  - [Code review](#code-review)
-  - [Styleguide](#styleguide)
-  - [Design guideline](#design-guideline)
-  - [API v1](#api-v1)
-  - [Developer Certificate of Origin (DCO)](#developer-certificate-of-origin-dco)
-  - [Release Cycle](#release-cycle)
-  - [Maintainers](#maintainers)
-  - [Owners](#owners)
-  - [Versions](#versions)
-  - [Releasing Gitea](#releasing-gitea)
-  - [Copyright](#copyright)
+<details><summary>Table of Contents</summary>
+
+- [Introduction](#introduction)
+- [Issues](#issues)
+  - [How to report issues](#how-to-report-issues)
+  - [Types of issues](#types-of-issues)
+  - [Discuss your design before the implementation](#discuss-your-design-before-the-implementation)
+- [Building Gitea](#building-gitea)
+- [Dependencies](#dependencies)
+  - [Backend](#backend)
+  - [Frontend](#frontend)
+- [Design guideline](#design-guideline)
+- [Styleguide](#styleguide)
+- [Copyright](#copyright)
+- [Testing](#testing)
+- [Translation](#translation)
+- [Code review](#code-review)
+  - [Pull request format](#pull-request-format)
+  - [PR title and summary](#pr-title-and-summary)
+  - [Milestone](#milestone)
+  - [Labels](#labels)
+  - [Breaking PRs](#breaking-prs)
+  - [Maintaining open PRs](#maintaining-open-prs)
+  - [Getting PRs merged](#getting-prs-merged)
+  - [Final call](#final-call)
+  - [Commit messages](#commit-messages)
+- [Documentation](#documentation)
+- [API v1](#api-v1)
+  - [GitHub API compatability](#github-api-compatability)
+  - [Adding/Maintaining API routes](#addingmaintaining-api-routes)
+  - [When to use what HTTP method](#when-to-use-what-http-method)
+  - [Requirements for API routes](#requirements-for-api-routes)
+- [Backports and Frontports](#backports-and-frontports)
+  - [What is backported?](#what-is-backported)
+  - [How to backport?](#how-to-backport)
+  - [Format of backport PRs](#format-of-backport-prs)
+  - [Frontports](#frontports)
+- [Developer Certificate of Origin (DCO)](#developer-certificate-of-origin-dco)
+- [Release Cycle](#release-cycle)
+- [Maintainers](#maintainers)
+- [TOC](#technical-oversight-committee-toc)
+  - [Current TOC members](#current-toc-members)
+  - [Previous TOC/owners members](#previous-tocowners-members)
+- [Governance Compensation](#governance-compensation)
+- [TOC \& Working groups](#toc--working-groups)
+- [Roadmap](#roadmap)
+- [Versions](#versions)
+- [Releasing Gitea](#releasing-gitea)
+
+</details>
+
```

Comment:
Do we need a TOC? GitHub does auto-generate these since a while now, but of course Gitea is still lacking that 😆.

Reply:
I mean, it's better than not having it, and especially as it is hidden by default it doesn't even waste any space when not needed.